lib/sender.js

```diff
@@ -75,8 +75,7 @@ class Sender {
     if (payloadLength === 126) {
       target.writeUInt16BE(data.length, 2);
     } else if (payloadLength === 127) {
-      target.writeUInt32BE(0, 2);
-      target.writeUInt32BE(data.length, 6);
+      target.writeUIntBE(data.length, 4, 6);
```

Comment:
This looks ok but we need to keep `target.writeUInt32BE(0, 2);` or use `target.writeUInt16BE(0, 2);` to ensure that the first two bytes of the 8 bytes payload length are zeroed. Note the use of `Buffer.allocUnsafe()` above that returns uninitialized memory.

That said, it is extremely unlikely that anyone sends more that 4 GiB of data as a single message. No issues have been raised, that I remember of, in this regard since I am a maintainer of this library.

Reply:
I have not used the `allocUnsafe` method before.  When you create a buffer using `Buffer.alloc(10)` all indexes are populated with 0 values unless overwritten.